Fix taxonomy merging scripts and ouput

Darcy220606 opened the pull request:
This PR updates pandas dependencies issues and fixes the taxonomy merging issues that were reported by users. Especially the ARG hamronization merging step were many taxonomy lineages were 'incorrectly' missing!


## PR checklist

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/funcscan/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/funcscan _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [x] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [x] Output Documentation in `docs/output.md` is updated.
- [x] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

Darcy220606 commented:
nf-core-bot fix linting

jfy133 reviewed (approved):
I've not checked the python code, but looks OK except for missing CHANGELOG

jasmezz commented on bin/merge_taxonomy.py:
```diff
@@ -66,6 +66,15 @@
+        mmseqs_taxonomy (tsv): mmseqs output file per sample
+
+    Returns:
+        data frame: reformated tables
```
```suggestion
        data frame: reformatted tables
```

jasmezz commented on bin/merge_taxonomy.py:
```diff
@@ -152,32 +180,46 @@ def combgc_taxa(args):
 def hamronization_taxa(args):
-    merged_df = pd.DataFrame()
+    """_summary_
+    Merges Hamronization tool output with taxonomy information.
```
```suggestion
    Merges hAMRonization tool output with taxonomy information.
```

jasmezz commented on bin/merge_taxonomy.py:
```diff
@@ -152,32 +180,46 @@ def combgc_taxa(args):
 #########################################
 # FUNCTION: HAMRONIZATION
 #########################################
+# TODO : FIX THE MERGING in ARG pipeline
```
Can this line go since you fixed the taxonmy merging below? Or does something else need to be fixed?

jasmezz commented on CHANGELOG.md:
```diff
@@ -15,6 +15,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 - [#427](https://github.com/nf-core/funcscan/pull/427) Fixed the AMP reference database issues reported by users, due to non-ASCII characters. (by @darcy220606)
 - [#430](https://github.com/nf-core/funcscan/pull/430) Updated `rgi/main` module to fix incorrect variable name. (by @amizeranschi and @jasmezz)
+- [#435](https://github.com/nf-core/funcscan/pull/435) Fixed dependency errors within taxonomy merging scripts and updated the code and output for all three workflows. Bumped to version 0.1.1. (by @darcy220606)
```
```suggestion
- [#435](https://github.com/nf-core/funcscan/pull/435) Fixed dependency errors within taxonomy merging scripts, updated the code and output for all three workflows. Bumped to version 0.1.1. (by @darcy220606)
```

jasmezz reviewed (approved):
I also did not scrutinize the updated python script here.

Darcy220606 commented:
@jfy133 and @jasmezz As requested, a summary of changes of this script: 
- Updates pandas syntax to latest version when possible
- Removed unnecessary header manipulation as now all subworkflows gives `contig_id` in the final parsed tables
 - Added docstring for all functions to make it easier to maintain the script 
 - Fixed the bug in Hamronization merging function by extracting the sample_id and contig_id string from the corresponding columns. This is because Hamronization when parsing it adds prefixes and suffixes in the final table to the IDs.
**Note: Every time there are changes made to the output of the subworkflows parsed output tables always make sure that the right column headers are still uptodate in this merging script otherwise it will fail to merge taxonomy.**

jasmezz commented on bin/merge_taxonomy.py:
```diff
@@ -197,29 +239,46 @@ def hamronization_taxa(args):
+        # ensure strings
+        subset_tool['contig_id'] = subset_tool['contig_id'].astype(str)
+        subset_taxa['contig_id'] = subset_taxa['contig_id'].astype(str)
+        # rename columns to avoid droping of mutual ones
```
```suggestion
        # rename columns to avoid dropping of mutual ones
```

jasmezz reviewed (approved):
LGTM, you have a go from me :) Only thing, I didn't follow the exact row manipulation for the hamronization parsing. It's fine by me, I'll just check if the resulting output file looks correct.
